.gitlab-ci.yml

```diff
@@ -0,0 +1,65 @@
+# This file is a template, and might need editing before it works on your project.
+image: golang:1.11.4
+
+variables:
+  # Please edit to your GitLab project
+  REPO_NAME: github.com/mendersoftware/mender
+
+before_script:
+  - mkdir -p /go/src/$(dirname $REPO_NAME)/mender /go/src/_/builds
+  - cp -r $CI_PROJECT_DIR /go/src/$(dirname $REPO_NAME)
+  # - ln -s /go/src/$(dirname $REPO_NAME) /go/src/_/builds/mendersoftware
+  - cd $GOPATH/src/$REPO_NAME
+  - apt-get update && apt-get install -yyq liblzma-dev
+  - make get-tools
+
+stages:
+  - test
+  - build
+  - publish
+
+test:format:
+  stage: test
+  allow_failure: true
+  script:
+    - go fmt $(go list ./... | grep -v /vendor/)
+    - go vet $(go list ./... | grep -v /vendor/)
+    - go test -race $(go list ./... | grep -v /vendor/)
+
+test:checks:
+  stage: test
+  script:
+    - make extracheck
```

Comment:
make extracheck will do `go vet` and `go fmt`. I think it is better to remove the test:format stage and, if any, add the `go test -race` to the Makefile's test target